src/tls/https_server_factory.rs

```diff
@@ -0,0 +1,47 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use hyper::error::Error as HyperError;
+use hyper::net::{ HttpsListener, Openssl, Ssl };
+use hyper::server::Server;
+use iron::{ Protocol, ServerFactory };
+use std::net::SocketAddr;
+use std::sync::{ Arc, Mutex };
+
+use tls::certificate_manager::CertificateManager;
+use tls::ssl_context::SniSslContextProvider;
+
+pub struct SniServerFactory<S: Ssl + Clone + Send> {
+    ssl: S
+}
+
+impl SniServerFactory<Openssl> {
+    pub fn new(ssl: &mut CertificateManager) -> Self {
+        use tls::SslContextProvider;
```

Comment:
Not particularly, I think I did this so that the trait would only be available in this scope, but that doesn't add anything right now.